doc/specs/#653 - Quake Mode/#653 - Quake Mode.md

```diff
@@ -0,0 +1,629 @@
+window "to the tray icon". This is a bit like when you close the Teams window,
+but Teams is actually still running in the system tray, or the "notification
+area".
+
+![The Teams tray icon](tray-icon-000.png)
+
+_fig 1: an example of the Teams tray icon in the notification area_.
+
+When users want to be able to "minimize to the tray", they want:
+* The window to no longer appear on the taskbar
+* The window to no longer appear in the alt-tab order
+
+When minimized to the tray, it's almost as if there's no window for the Terminal
+at all. This can be combined with the global hotkey to quickly restore the window.
+
+The tray icon could be used for a variety of purposes. As a simple start, we
+could include the following three options:
+
+```
+Focus Terminal
+---
+Windows > Window 1 - <un-named window>
+          Window 2 - "This-window-does-have-a-name"
+---
+Quit
+```
+
+Just clicking on the icon would summon the recent terminal window. Right
+clicking would show the menu with "Focus Terminal", "Windows" and "Quit" in it, and
+"Windows" would have nested entries for each Terminal window.
+
+* "Focus Terminal" would do just that - summon the most recent terminal window,
+  wherever it is.
+* "Windows" would have nested popups for each open Terminal window. Each of
+  these nested entries would display the name and ID of the window. Clicking
+  them would summon that window (wherever it may be)
+* "Quit" would be akin to quit in browsers - close all open windows
+  <sup>[[1]](#footnote-1)</sup>.
+
+The tray notification would be visible always when the user has
+`"minimizeToTray": true` set in their settings. If the user has that set to
+false, but would still like the tray, they can specify `"alwaysShowTrayIcon":
+true`. That will cause the tray icon to always be added to the system tray.
+
+There's not a combination of settings where the Terminal is "minimized to the
+tray", and there's _no tray icon visible_. We don't want to let users get into a
+state where the Terminal is running, but is totally hidden from their control.
+
+From a technical standpoint, the tray icon is managed similar to the global
+hotkey. The Monarch process is responsible for setting it up, and processing the
+messages. When a Monarch dies and a new process becomes the Monarch, then it
+will re-create the tray icon.
+
+## UI/UX Design
+
+To summarize, we're proposing the following set of settings:
+
+```jsonc
+{
+    "minimizeToTray": bool,
+    "alwaysShowTrayIcon": bool,
+    "globalHotkeys": [
+        {
+            "keys": KeyChord,
+            "dropdownDuration": float,
+            "hideWhenVisible": bool,
+            "monitor": "any"|"toCurrent"|"onCurrent"|int,
+            "desktop": null|"toCurrent"|"onCurrent"
+        }
+    ]
+}
+```
+
+## Potential Issues
+
+<table>
+
+<tr>
+<td><strong>Compatibility</strong></td>
+<td>
+
+As part of this set of changes, we'll also be allowing the <kbd>Win</kbd> key in

```

Comment:
Maybe I'm missing something, but I don't quite get it why <kbd>Win</kbd> is required for Quake mode. I've been using <kbd>Ctrl + `</kbd> since like forever.

Reply:
Oh <kbd>Win</kbd> definitely _isn't_ required. It's just a nice feature that'll be coming with this. I'll update the samples so that some are using other keys too

Reply:
It also would be nice to include Windows Terminal as an option for Touchpad actions.
e.g. imagine if you could configure the terminal to slide down when swipe down with 3 or 4 fingers

![image](https://user-images.githubusercontent.com/1232175/110228102-62d34680-7ef6-11eb-9311-eb5ec8ef7e0c.png)

Reply:
bonus points if this behaviour is automatically configured when you enable "Developer mode"

Reply:
Unfortunately, I don't think apps can register some sort of action with the OS like that. That would be cool, but unfortunately impossible 😕